web-app/frontend/src/pages/history/history.tsx

```diff
@@ -1,8 +1,93 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
 import styles from "./history.module.scss";
+import Card from "../../components/pageContent/card/card";
+import Plot from "react-plotly.js";
+import typedUseSelector from "../../redux/reduxInterfaces";
+import TitleBox from "../../components/pageContent/titleBox/titleBox";
+import { makePlotlyLayout } from "../dataColleciton/dataCollection";
+import { PlotData } from "plotly.js";
 
-const history: React.FC = () => {
-  return <div className={styles.wrapper}>History!!!</div>;
+const initialGraphState = () => {
+  return {
+    x: [new Date().toTimeString().split(" ")[0]],
+    y: [0],
+    type: "scatter"
+  };
 };
 
-export default history;
+const bundleGraph = (graphObject: object[]) => {
+  let graph = {
+    x: [],
+    y: [],
+    type: "scatter"
+  };
+  graphObject.forEach((dataPoint: any) => {
+    graph = {
+      ...graph,
+      x: graph.x.concat(dataPoint.x),
+      y: graph.y.concat(dataPoint.y)
+    };
+  });
+  return graph;
+};
+
+const History: React.FC = () => {
+  const [ids, updateIds] = useState([0]);
+
+  const [graphData, updateGraph] = useState(initialGraphState() as any);
+
+  const getData = async () => {
+    const responseData = await fetch("http://localhost:4000/history/GET/ids", {
+      method: "GET",
+      mode: "cors",
+      headers: {
+        "Content-Type": "application/json"
+      }
+    }).then(response => response.json());
+    updateIds(responseData);
+  };
+
+  useEffect(() => {
+    getData();
+  }, []);
+
+  const handleCardClick = async (id: number) => {
+    const data = await fetch(`http://localhost:4000/history/GET/${id}/data`, {
```

Comment:
That's a really good point! I'm a big fan of await, as it removes all of the ambiguity with promises, as you know the async code will not move forward until a promise has returned. 

So in the example, you gave there where you call `bundleGraph(response.json())` this has the potential of getting messed up as `.json()` returns a promise to a json, not explicitly a json. Using the await ensures that cardData will always have a value, and not a promise before it progresses to the next line.

I'm a really big fan of async/await instead of chaining `.then`s. It makes it much easier to debug too. Instead of needing to place a debug in each .then, you can simply step through it as if it wasn't async code!! async/await is honestly one of my favorite things in javascript 
😄